File: HollowGoodsLibrary/src/main/java/com/hg/hollowgoods/Adapter/BaseRecyclerView/CommonAdapter.java
package com.hg.hollowgoods.Adapter.BaseRecyclerView;

import android.content.Context;

import com.hg.hollowgoods.Adapter.BaseRecyclerView.Base.ItemViewDelegate;
import com.hg.hollowgoods.Adapter.BaseRecyclerView.Base.ViewHolder;

import java.util.List;

/**
 * Created by HG
 */
public abstract class CommonAdapter<T> extends MultiItemTypeAdapter<T> {

    public CommonAdapter(final Context context, final int layoutId, List<T> data) {
        super(context, data);
        addItemViewDelegate(new ItemViewDelegate<T>() {
            @Override
            public int getItemViewLayoutId() {
                return layoutId;
            }

            @Override
            public boolean isForViewType(T item, int position) {
                return true;
            }

            @Override
            public void convert(ViewHolder viewHolder, T item, int position) {
                CommonAdapter.this.convert(viewHolder, item, position);
            }
        });
    }

    protected abstract void convert(ViewHolder viewHolder, T item, int position);

}

File: HollowGoodsLibrary/src/main/java/com/hg/hollowgoods/Adapter/BaseAdapterView/CommonAdapter.kt
package com.hg.hollowgoods.Adapter.BaseAdapterView

import android.content.Context
import com.hg.hollowgoods.Adapter.BaseAdapterView.Base.ItemViewDelegate

/**
 * AdapterView公共适配器
 * Created by Hollow Goods on 2018-03-22.
 */
abstract class CommonAdapter<T>(context: Context, layoutId: Int, data: List<T>) : MultiItemTypeAdapter<T>(context, data) {

    init {
        this.addItemViewDelegate(object : ItemViewDelegate<T> {
            override fun getItemViewLayoutId(): Int {
                return layoutId
            }

            override fun isForViewType(item: T, position: Int): Boolean {
                return true
            }

            override fun convert(viewHolder: ViewHolder, t: T, position: Int) {
                this@CommonAdapter.convert(viewHolder, t, position)
            }
        })
    }

    abstract override fun convert(viewHolder: ViewHolder, item: T, position: Int)

}